azuredevops/internal/service/serviceendpoint/commons.go

```diff
@@ -240,3 +276,29 @@ func genServiceEndpointDeleteFunc(expandFunc expandFunc) schema.DeleteFunc {
 		return deleteServiceEndpoint(clients, projectID, serviceEndpoint.Id)
 	}
 }
+
+func getServiceEndpoint(client *client.AggregatedClient, serviceEndpointID *uuid.UUID, projectID *string) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		serviceEndpoint, err := client.ServiceEndpointClient.GetServiceEndpointDetails(
+			client.Ctx,
+			serviceendpoint.GetServiceEndpointDetailsArgs{
+				EndpointId: serviceEndpointID,
+				Project:    projectID,
+			},
+		)
+
+		if err != nil {
+			return nil, opState.Failed, fmt.Errorf("Error looking up service endpoint given ID (%v) and project ID (%v): %v ", serviceEndpointID, projectID, err)
+		}
+
+		if *serviceEndpoint.IsReady {
+			return serviceEndpoint, opState.Ready, nil
+		} else if serviceEndpoint.OperationStatus != nil {
+			opStatus := (serviceEndpoint.OperationStatus).(map[string]interface{})
+			if opStatus["state"] == opState.Failed {
+				return nil, opState.Failed, fmt.Errorf("Error looking up service endpoint given ID (%v) and project ID (%v): %v ", serviceEndpointID, projectID, serviceEndpoint.OperationStatus)
```

Comment:
This error message is the same as on line 291, should this text be placed in a variable?